src/idl_gen_text.cpp

```diff
@@ -263,8 +263,9 @@ bool GenerateText(const Parser &parser, const void *flatbuffer,
   std::string &text = *_text;
   assert(parser.root_struct_def_);  // call SetRootType()
   text.reserve(1024);               // Reduce amount of inevitable reallocs.
-  if (!GenStruct(*parser.root_struct_def_, GetRoot<Table>(flatbuffer), 0,
-                 parser.opts, _text)) {
+  auto root = parser.opts.size_prefixed ?
+      GetSizePrefixedRoot<Table>(flatbuffer) : GetRoot<Table>(flatbuffer);
+  if (!GenStruct(*parser.root_struct_def_, root, 0, parser.opts, _text)) {
```

Comment:
This reads size prefixed. Code for writing size prefixed seems missing.

Reply:
In my PR I have missed this part. Then it should also probably be included in `GenFieldOffset`, right?

Reply:
I've added the code to write also.
@robert-schmidtke, I don't think it applies to GenFieldOffset as the size prefix is only for  the top root table, if I had understood correctly.

Reply:
So nested flatbuffers never have their size prefixed?

Reply:
@aardappel @desqaz should there be two distinct options, one for reading size prefixed binaries, and one for writing them? or should reading size prefixed input always also entail writing size prefixed output?

Reply:
I'd vote for one option. I don't see a lot of reasons why 2 would be better.

Reply:
There is no reason for a nested_flatbuffer to have a size prefix, since its length is known from the vector it sits in.